ledger/ledger-on-sql/src/main/scala/com/daml/ledger/on/sql/SqlLedgerReaderWriter.scala

```diff
@@ -112,19 +132,30 @@ object SqlLedgerReaderWriter {
           updateOrRetrieveLedgerId(ledgerId, database).removeExecutionContext
         )
         dispatcher <- new DispatcherOwner(database).acquire()
-        validator = SubmissionValidator.createForTimeMode(
-          ledgerStateAccess = new SqlLedgerStateAccess(database, metrics),
-          logEntryIdAllocator = logEntryIdAllocator,
-          checkForMissingInputs = false,
-          stateValueCache = stateValueCache,
-          engine = engine,
-          metrics = metrics,
-          inStaticTimeMode = timeProvider != TimeProvider.UTC,
+        sqlLedgerStateAccess = new SqlLedgerStateAccess(database, metrics)
+        keyValueCommitting = new KeyValueCommitting(
+          engine,
+          metrics,
         )
-        committer = new ValidatingCommitter[Index](
-          () => timeProvider.getCurrentTime,
-          validator = validator,
-          postCommit = dispatcher.signalNewHead,
+        ledgerDataExporter <- LedgerDataExporter.Owner.acquire()
+        committer = new Committer(
+          transformStateReader = ledgerStateReader =>
+            CachingStateReader(
+              stateValueCache,
+              ImmutablesOnlyCacheUpdatePolicy,
+              DamlLedgerStateReader
+                .from(ledgerStateReader, StateKeySerializationStrategy.createDefault()),
```

Comment:
Please inject the `keySerializationStrategy` instead of constructing it here (and below).